File: Trinity.Runtime/platform_src/ios/elastOS/Classes/PasswordManager/PasswordInfo/BankAccountPasswordInfo.swift
/*
* Copyright (c) 2020 Elastos Foundation
*
* Permission is hereby granted, free of charge, to any person obtaining a copy
* of this software and associated documentation files (the "Software"), to deal
* in the Software without restriction, including without limitation the rights
* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
* copies of the Software, and to permit persons to whom the Software is
* furnished to do so, subject to the following conditions:
*
* The above copyright notice and this permission notice shall be included in all
* copies or substantial portions of the Software.
*
* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
* SOFTWARE.
*/

import Foundation

public class BankAccountPasswordInfo : PasswordInfo {
    var accountOwner: String? = nil
    var iban: String? = nil
    var swift: String? = nil
    var bic: String? = nil

    public static func fromJsonObject(_ jsonObject: Dictionary<String, Any>) throws -> PasswordInfo {
        let info = BankAccountPasswordInfo()

        try info.fillWithJsonObject(jsonObject)

        return info;
    }

    public override func asDictionary() -> Dictionary<String, Any>? {
        if var jsonObject = super.asDictionary() {
            jsonObject["accountOwner"] = accountOwner
            jsonObject["iban"] = iban
            jsonObject["swift"] = swift
            jsonObject["bic"] = bic

            return jsonObject
        }
        
        return nil
    }

    public override func fillWithJsonObject(_ jsonObject: Dictionary<String, Any>) throws {
        // Fill base fields
        try super.fillWithJsonObject(jsonObject)

        // Fill specific fields
        if (jsonObject.keys.contains("accountOwner")) {
            self.accountOwner = jsonObject["accountOwner"] as? String
        }
        if (jsonObject.keys.contains("iban")) {
            self.iban = jsonObject["iban"]  as? String
        }
        if (jsonObject.keys.contains("swift")) {
            self.swift = jsonObject["swift"]  as? String
        }
        if (jsonObject.keys.contains("bic")) {
            self.bic = jsonObject["bic"]  as? String
        }
    }
}

File: Trinity.Runtime/platform_src/android/app/src/main/java/org/elastos/trinity/runtime/passwordmanager/passwordinfo/BankAccountPasswordInfo.java
package org.elastos.trinity.runtime.passwordmanager.passwordinfo;

import org.json.JSONException;
import org.json.JSONObject;

public class BankAccountPasswordInfo extends PasswordInfo {
    String accountOwner = null;
    String iban = null;
    String swift = null;
    String bic = null;

    public static PasswordInfo fromJsonObject(JSONObject jsonObject) throws Exception {
        BankAccountPasswordInfo info = new BankAccountPasswordInfo();

        info.fillWithJsonObject(jsonObject);

        return info;
    }

    @Override
    public JSONObject asJsonObject() {
        try {
            JSONObject jsonObject = super.asJsonObject();

            jsonObject.put("accountOwner", accountOwner);
            jsonObject.put("iban", iban);
            jsonObject.put("swift", swift);
            jsonObject.put("bic", bic);

            return jsonObject;
        } catch (
                JSONException e) {
            e.printStackTrace();
            return null;
        }
    }

    @Override
    public void fillWithJsonObject(JSONObject jsonObject) throws Exception {
        // Fill base fields
        super.fillWithJsonObject(jsonObject);

        // Fill specific fields
        if (jsonObject.has("accountOwner")) {
            this.accountOwner = jsonObject.getString("accountOwner");
        }
        if (jsonObject.has("iban")) {
            this.iban = jsonObject.getString("iban");
        }
        if (jsonObject.has("swift")) {
            this.swift = jsonObject.getString("swift");
        }
        if (jsonObject.has("bic")) {
            this.bic = jsonObject.getString("bic");
        }
    }
}
